public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -176,16 +185,36 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
       .map(target => this.prepareLiveTarget(target, options));
 
     for (const liveTarget of liveTargets) {
-      const subscription = webSocket(liveTarget.url)
+      liveTarget.subscription = webSocket(liveTarget.url)
         .pipe(
-          map((results: any[]) => {
-            const delta = this.processResult(results, liveTarget);
+          map((response: any) => {
+            const { refId } = liveTarget;
+
+            // What should we do with:
+            // response.dropped_entries?
+
+            const streams: LokiLogsStream[] = response.streams;
+            if (streams) {
+              for (const stream of streams) {
```

Comment:
The buffer is per stream not sure if that is useful for the live stream usecase. @hugohaggmark correct me here if I am wrong but we basically merge all the streams from loki into one and show them together (I think this should happen on Loki end but it does not seem to be always the case so we do it on our end also). So the buffer here should be global across all streams I think to be useful for the live stream usecase.

Reply:
Yes. This PR doesn't actually take care of buffering data for a period of time or have I missed it?

Reply:
This does not take care of buffering for a time period -- it is just a cap on the size buffer.  If we want a cap on the time-period, I think we still need a max size so we don't accidentally get too big.